### PR TITLE
add service monitor

### LINKED ISSUE #41: Add support prometheus-operator?
This will simplify the process of collecting Prometheus metrics

### PR DESCRIPTION
**Pull request by batazor**
**What type of PR is this?**

**Which issue does this PR fix**:
close #41

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
@batazor can you pls take a look at the CI failures?

**Comment by batazor** [comment]
> @batazor can you pls take a look at the CI failures?

done

**Comment by rchincha** [comment]
@batazor upgrade test failed as follows.

 Get the application URL by running these commands:
  export NODE_PORT=$(kubectl get --namespace zot-e817pun8up -o jsonpath="{.spec.ports[0].nodePort}" services zot-e817pun8up)
  export NODE_IP=$(kubectl get nodes --namespace zot-e817pun8up -o jsonpath="{.items[0].status.addresses[0].address}")
  echo http://$NODE_IP:$NODE_PORT
Error: UPGRADE FAILED: template: zot/templates/servicemonitor.yaml:1:18: executing "zot/templates/servicemonitor.yaml" at <.Values.metrics.enabled>: nil pointer evaluating interface {}.enabled

**Comment by batazor** [comment]
@rchincha Yes, I added support case handling when `values.yaml` was not set. Thanks!

File: charts/zot/Chart.yaml [hunk 1]
```diff
@@ -3,4 +3,4 @@ appVersion: v2.1.0-rc2
 description: A Helm chart for Kubernetes
 name: zot
 type: application
-version: 0.1.56
+version: 0.1.57
```
Comment (rchincha) [inline charts/zot/Chart.yaml:6]: This needs a version bump to 0.1.58
We just released a newer zot version

Comment (batazor) [reply charts/zot/Chart.yaml]: done

**Comment by rchincha** [comment]
@batazor Still fails.

```
 Get the application URL by running these commands:
  export NODE_PORT=$(kubectl get --namespace zot-ue4ohhrx84 -o jsonpath="{.spec.ports[0].nodePort}" services zot-ue4ohhrx84)
  export NODE_IP=$(kubectl get nodes --namespace zot-ue4ohhrx84 -o jsonpath="{.items[0].status.addresses[0].address}")
  echo http://$NODE_IP:$NODE_PORT
Error: UPGRADE FAILED: template: zot/templates/servicemonitor.yaml:1:34: executing "zot/templates/servicemonitor.yaml" at <.Values.metrics.enabled>: nil pointer evaluating interface {}.enabled
===================================
```

**Comment by batazor** [comment]
@rchincha Okay, one more try, added even more complete variable checking

**Comment by rchincha** [comment]
@batazor 

https://github.com/project-zot/helm-charts/pull/43
^ had to pick whatever was approved and ready.
Your PR may require one more version bump.

**Comment by batazor** [comment]
@rchincha  no problem

**Comment by rchincha** [comment]
@batazor why close this PR?

**Comment by batazor** [comment]
I don't understand the problems that appear during testing, I haven't experienced such problems locally

**Pull request closed without merging**